src/common/segmented_json.h

```diff
@@ -85,6 +86,28 @@ class ChunkRenderer {
     virtual std::tuple<JsonResult, size_t> render(uint8_t *buffer, size_t buffer_size) = 0;
 };
 
+/// Renders nothing, can act as a placeholder when composing stuff together.
+class EmptyRenderer : public ChunkRenderer {
+public:
+    virtual std::tuple<JsonResult, size_t> render(uint8_t *, size_t) override;
+};
+
+template <class... V>
+class VariantRenderer : public ChunkRenderer {
+private:
+    std::variant<V...> data;
+
+public:
+    VariantRenderer() = default;
+    VariantRenderer(std::variant<V...> d)
```

Comment:
:thinking: why do you pass the parameter by value and then move it into `data`?
Usually the parameter is an r-value reference in similar cases.

Reply:
Fixed. Probably a leftover from fights with different kinds of C++ references, templates (that were involved more in some intermediate version), failures with `std::forward`… sometimes C++ goes off the rail and this had the benefit of actually being able to compile 😈

Reply:
I'd like to point out that the previous version was also fine, pass-by-value and then std::move is generally a preferred way of doing setters, rather than having to specify multiple overloads (`const T &` and `T &&`). Performance wise it seems to be about the same, per https://stackoverflow.com/a/10706636